xbmc/platform/android/activity/JNIXBMCConnectivityManagerNetworkCallback.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright (C) 2012-2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "JNIXBMCConnectivityManagerNetworkCallback.h"
+
+#include "CompileInfo.h"
+
+#include <androidjni/Context.h>
+#include <androidjni/jutils-details.hpp>
+
+using namespace jni;
+
+static std::string s_className =
+    std::string(CCompileInfo::GetClass()) + "/interfaces/XBMCConnectivityManagerNetworkCallback";
+
+CJNIXBMCConnectivityManagerNetworkCallback::CJNIXBMCConnectivityManagerNetworkCallback()
+  : CJNIBase(s_className)
+{
+  m_object = new_object(CJNIContext::getClassLoader().loadClass(GetDotClassName(s_className)));
+  m_object.setGlobal();
+
+  add_instance(m_object, this);
+}
+
+CJNIXBMCConnectivityManagerNetworkCallback::~CJNIXBMCConnectivityManagerNetworkCallback()
+{
+  remove_instance(this);
+}
+
+void CJNIXBMCConnectivityManagerNetworkCallback::RegisterNatives(JNIEnv* env)
+{
+  jclass cClass = env->FindClass(s_className.c_str());
```

Comment:
const?

Reply:
I think I would leave it like this, I don't know if it affects JNI calls and there is no `const jclass` declaration in the code